File: Lab 9/t4.go
package main

import (
  "fmt"
  "os"
  "bufio"
  "strings"
)

func main() {
  var (
    buf []byte // For ReadLine use
    e error
  )

  stdin := bufio.NewReader(os.Stdin)

  fmt.Println("Please input the first string:")
  buf, _, e = stdin.ReadLine()
  if e != nil {
    fmt.Println("Read error:", e)
    return
  }
  str1 := string(buf)

  stdin.Discard(stdin.Buffered())

  fmt.Println("Please input the second string:")
  buf, _, e = stdin.ReadLine()
  if e != nil {
    fmt.Println("Read error:", e)
    return
  }
  str2 := string(buf)

  if strings.Contains(str2, str1) {
    fmt.Println("The first string is a substring of the second.")
  } else {
    fmt.Println("The first string is not a substring of the second.")
  }
}

File: Lab 9/t4.rs
use std::io::stdin;
use std::process::exit;

fn main() {
  println!("Please input the first string:");
  let mut str1 = String::new();
  match stdin().read_line(&mut str1) {
    Ok(_) => {},
    Err(e) => {
      eprintln!("Read error: {}", e);
      exit(1);
    }
  }
  // Strip newline, then shadow it (becoming immutable)
  let str1 = str1.trim_right();

  println!("Please input the second string:");
  let mut str2 = String::new();
  match stdin().read_line(&mut str2) {
    Ok(_) => {},
    Err(e) => {
      eprintln!("Read error: {}", e);
      exit(1);
    }
  }
  // Strip newline, then shadow it (becoming immutable)
  let str2 = str2.trim_right();

  println!("The first string is{} a substring of the second.",
           if str2.contains(str1) { "" } else { " not" });
}
